specification/frontdoor/resource-manager/Microsoft.Network/stable/2020-08-01/frontdoor.json

```diff
@@ -1,2358 +1,2398 @@
+                        "description": "A health probe settings",
+                        "$ref": "#/definitions/HealthProbeSettingsModel"
+                    }
+                },
+                "backendPools": {
+                    "description": "Backend pools available to routing rules.",
+                    "type": "array",
+                    "items": {
+                        "description": "A backend pool",
+                        "$ref": "#/definitions/BackendPool"
+                    }
+                },
+                "frontendEndpoints": {
+                    "description": "Frontend endpoints available to routing rules.",
+                    "type": "array",
+                    "items": {
+                        "description": "A frontend endpoint",
+                        "$ref": "#/definitions/FrontendEndpoint"
+                    }
+                },
+                "backendPoolsSettings": {
+                    "description": "Settings for all backendPools",
+                    "$ref": "#/definitions/BackendPoolsSettings"
+                },
+                "enabledState": {
+                    "description": "Operational status of the Front Door load balancer. Permitted values are 'Enabled' or 'Disabled'",
+                    "enum": [
+                        "Enabled",
+                        "Disabled"
+                    ],
+                    "type": "string",
+                    "x-ms-enum": {
+                        "name": "FrontDoorEnabledState",
+                        "modelAsString": true
+                    }
+                },
+                "webApplicationFirewallPolicyLinks": {
+                    "description": "Defines the Web Application Firewall links (if applicable)",
+                    "type": "array",
+                    "items": {
+                        "description": "Defines the Web Application Firewall link",
+                        "type": "object",
+                        "properties": {
+                            "id": {
+                                "description": "Resource ID.",
+                                "type": "string"
+                            },
+                            "patterns": {
+                                "description": "List of patterns to match.",
+                                "type": "array",
+                                "items": {
+                                    "description": "Patterns to match.",
+                                    "type": "object",
+                                    "properties": {
+                                        "domains": {
+                                            "description": "Domains to match.",
+                                            "type": "array",
+                                            "items": {
+                                                "description": "Domains.",
+                                                "type": "string"
```

Comment:
The type here may need to be a resource id reference to frontend endpoints. It's less error prone. The one issue is that there would be no wildcard support, but that might be fine.

Reply:
We need to support wildcards though so I think that could be an issue.